File: Server/Server/Demo/logic/LoginMsgHandle.cs
using System;
using YT;

namespace YT

{
    public partial class MsgHandler
    {


        //注册协议处理
        public static void MsgRegister(Connection c, MsgBase msgBase) {
            MsgRegister msg = (MsgRegister)msgBase;
            //注册
            if (DBManager.Register(msg.id, msg.pw)) {
                DBManager.CreatePlayer(msg.id);
                msg.result = 0;
            }
            else {
                msg.result = 1;
            }
            c.Send(msg);
        }


        //登陆协议处理
        public static void MsgLogin(Connection c, MsgBase msgBase) {

            MsgLogin msg = (MsgLogin)msgBase;
            //密码校验
            if (!DBManager.CheckPassword(msg.id, msg.pw)) {
                msg.result = 1;
                c.Send(msg);
                return;
            }
            //不允许再次登陆
            if (c.player != null) {
                msg.result = 1;
                c.Send(msg);
                return;
            }
            //如果已经登陆，踢下线
            if (PlayerManager.IsOnline(msg.id)) {
                //发送踢下线协议
                Player other = PlayerManager.GetPlayer(msg.id);
                MsgKick msgKick = new MsgKick();
                msgKick.reason = 0;
                other.Send(msgKick);
                //断开连接
                c.Send(msg);
            }
            //获取玩家数据
            PlayerData playerData = DBManager.GetPlayerData(msg.id);
            if (playerData == null) {
                msg.result = 1;
                c.Send(msg);
                return;
            }
            //构建Player
            Player player = new Player(c);
            player.id = msg.id;
            player.data = playerData;
            PlayerManager.AddPlayer(msg.id, player);
            c.player = player;
            //返回协议
            msg.result = 0;
            player.Send(msg);
        }
    }
}

File: Server-Go/Demo/MsgHandler/LoginMsgHandle.go
package MsgHandler

import (
	"ReaminHistory/Demo/DB"
	Player2 "ReaminHistory/Demo/Player/Player"
	"ReaminHistory/Demo/Player/PlayerManager"
	"ReaminHistory/Demo/Proto"
	"fmt"
	"github.com/golang/protobuf/proto"
	"github.com/mogebingxue/YTFramework/Net"
)

//注册协议处理
func MsgRegister(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgRegister{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if DB.Register(msg.Id, msg.Pw) {
		DB.CreatePlayer(msg.Id)
		msg.Result = 0
	} else {
		msg.Result = 1
	}
	fmt.Println()
	c.Send(msg)
}

//登陆协议处理
func MsgLogin(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgLogin{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	//密码校验
	if !DB.CheckPassword(msg.Id, msg.Pw) {
		msg.Result = 1
		c.Send(msg)
		return
	}
	//不允许再次登录
	if _, ok := PlayerManager.Players[c.Conv]; ok {
		msg.Result = 1
		c.Send(msg)
		return
	}
	//如果已经登陆，踢下线
	if PlayerManager.IsOnline(msg.Id) {
		msgKick := &Proto.MsgKick{}
		msgKick.Result = 0
		PlayerManager.GetPlayer(msg.Id).Send(msgKick)
	}
	playerData := DB.GetPlayerData(msg.Id)
	if playerData == nil {
		msg.Result = 1
		c.Send(msg)
		return
	}
	player := Player2.NewPlayer(c.Conv)
	player.Id = msg.Id
	player.Data = playerData
	PlayerManager.AddPlayer(msg.Id, player)
	if PlayerManager.Players == nil {
		PlayerManager.Players = make(map[uint32]*Player2.Player)
	}
	PlayerManager.Players[c.Conv] = player
	msg.Result = 0
	fmt.Println("id:", msg.Id, "pw:", msg.Pw, "result:", msg.Result)
	player.Send(msg)
}
